src/main/java/org/cactoos/scalar/CheckedScalar.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2018 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.scalar;
+
+import org.cactoos.Func;
+import org.cactoos.Scalar;
+import org.cactoos.func.UncheckedFunc;
+
+/**
+ * Scalar that throws exception specified by user.
+ *
+ *  <p>There is no thread-safety guarantee.
+ *
+ * @author Vedran Vatavuk (123vgv@gmail.com)
+ * @version $Id$
+ * @param <T> Type of result
+ * @param <E> Type of exception
+ * @since 0.30
+ */
+public final class CheckedScalar<T, E extends Exception> implements Scalar<T> {
+
+    /**
+     * Function that wraps exception.
+     */
+    private final Func<Exception, E> func;
```

Comment:
I believe `wrapper` would be the name exactly explains the purpose of this field. But I know about "no -er" rule... Damn :) Still `func` is not the best name. Could you try to find any better word?

Reply:
@proshin-roman I changed the name to `wrapped` although there are many classes that use `func` as a field variable name. For example: `SolidScalar`, `StickyScalar`, `RetryScalar`...